src/sdam/topology.rs

```diff
@@ -53,146 +52,56 @@ use super::{
 pub(crate) struct Topology {
     watcher: TopologyWatcher,
     updater: TopologyUpdater,
-    check_requester: TopologyCheckRequester,
+    check_requester: TopologyMonitorsManager,
     _worker_handle: WorkerHandle,
 }
 
 impl Topology {
     pub(crate) fn new(options: ClientOptions) -> Result<Topology> {
         let http_client = HttpClient::default();
-        let description = TopologyDescription::new(options.clone())?;
-        let is_load_balanced = options.load_balanced == Some(true);
+        let description = TopologyDescription::default();
 
-        let update_requester = TopologyCheckRequester::new();
+        let update_requester = TopologyMonitorsManager::new();
         let event_emitter = options.sdam_event_handler.as_ref().map(|handler| {
             let (tx, mut rx) = mpsc::unbounded_channel::<AcknowledgedMessage<SdamEvent>>();
 
+            // Spin up a task to handle events so that a user's event handling code can't block the
+            // TopologyWorker.
             let handler = handler.clone();
             runtime::execute(async move {
                 while let Some(event) = rx.recv().await {
                     let (event, ack) = event.into_parts();
-
-                    let is_closed = matches!(event, SdamEvent::TopologyClosed(_));
                     handle_sdam_event(handler.as_ref(), event);
-
                     ack.acknowledge(());
-
-                    // no more events should be emitted after a TopologyClosedEvent
-                    if is_closed {
-                        break;
-                    };
                 }
             });
             SdamEventEmitter { sender: tx }
         });
 
         let (updater, update_receiver) = TopologyUpdater::channel();
         let (worker_handle, handle_listener) = WorkerHandleListener::channel();
-
```

Comment:
We used to do a lot of ad-hoc initialization to match the events and outcomes that were expected in the SDAM tests, but it was pretty hard to follow / maintain. I've updated this to use the standard topology update process that all other updates go through, which cuts down on a lot of this hard-coded stuff.